File: OCH_Tao/ABC/B/159.py
S = input()
N = len(S)
if S==S[::-1]:
  if (X:=S[:(N-1)//2])==X[::-1]:
    if (Y:=S[(N+3)//2-1:])==Y[::-1]:
      print("Yes")
    else:
      print("No")
  else:
    print("No")
else:
  print("No")

File: OCH_Tao/ABC/B/159.js
const S = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n")[0];
const N = S.length;
if (S === [...S].reverse().join("")) {
  if (S.slice(0, (N - 1) / 2) === [...S.slice(0, (N - 1) / 2)].reverse().join("")) {
    if (S.slice((N + 3) / 2 - 1) === [...S.slice((N + 3) / 2 - 1)].reverse().join("")) {
      console.log("Yes");
    } else {
      console.log("No");
    }
  } else {
    console.log("No");
  }
} else {
  console.log("No");
}